blogposts/2021/continuous_delivery.md

```diff
@@ -0,0 +1,174 @@
+
+The purpose of continuous deployment, then, in Fitz’s definition and the current definition, is to make deployment scalable.
+
+Fitz was perhaps more right than he realized. Traditional deployment practices have only gotten less practical and less scalable.
+
+In our research (see the [Big code report](https://info.sourcegraph.com/emergence-of-big-code-2020-survey?_ga=2.117030534.1813537383.1623859682-2046487521.1621427028)), we found that 51% of respondents reported having more than 100 times the volume of code they had 10 years ago and 18% saying they have 500 times more code.
+
+As Fitz pointed out 12 years ago, so-called “stupid mistakes [...] happen more frequently as the product grows more complex.” But they’re _not_ stupid mistakes, we need to point out; if something is inevitable and consistent, it’s not an individual mistake–it’s a flaw in the system.
+
+That’s why Fitz proposed continuous deployment as a new system. If you’re deploying continuously, he writes, then you can fail fast. He writes that “the closer a failure is to the point where it was introduced, the more data you have to correct for that failure.”
+
+If Alex had deployed as soon as she developed the change, the failure would have been less severe, the rollback faster, and the diagnosis easier to make. And in a puff of smoke, theoretically, goes release anxiety.
+
+### True continuous deployment is impossible for all
+
+The problem with continuous deployment is that it requires a SaaS environment. If you deploy your product as a cloud-based service, then you can push instantly available changes all day long.
+
+But that’s not the reality for most companies. Despite the seeming dominance of SaaS, on-prem and self-hosted environments are not only still alive but thriving.
+
+According to a [Replicated and Dimensional Research study](https://www.replicated.com/blog/on-prem-sales-and-demand-are-rising/): 92% of companies indicate on-premises software sales are growing. That’s right, not declining or stabilizing–_growing_. In fact, their research shows customer demand for on-premises software actually equals that for public cloud. Not only is on-premises growing, but according to their research, 91% of customers report being satisfied with their on-premises software.
+
+So, let’s restate that:
+
+- More and more companies are selling and deploying on-premises software;
+- Demand for on-premises software matches demand for the seemingly much hotter market of cloud-based software; and
+- Most companies are happy with their on-premises software.
+
+This narrative violation isn’t the result of split camps, either. It’s not as though we have one camp using the public cloud and one camp sticking with on-prem. (Sourcegraph, alone, disproves this with our products being split among self-hosted and cloud environments).
```

Comment:
Is there anywhere we can link to (on the about site/handbook) that explains how our product is split?